virtool/subtractions/oas.py

```diff
@@ -0,0 +1,20 @@
+from typing import Optional
+from pydantic import BaseModel, Field, constr
+
+
+class EditSubtractionSchema(BaseModel):
+    """
+    Used when modifying a Subtraction
+    """
+    name: Optional[constr(strip_whitespace=True, min_length=1)] = Field(description="A unique name for the host (eg. "
+                                                                                    "Arabidopsis)")
+    nickname: Optional[constr(strip_whitespace=True)] = Field(description="A nickname of the host")
+
+
+class CreateSubtractionSchema(BaseModel):
+    """
+    Used for creating a new Subtraction.
+    """
+    name: constr(strip_whitespace=True, min_length=1) = Field(description="A unique name for the host (eg. Arabidopsis)")
+    nickname: constr(strip_whitespace=True) = Field(description="A nickname of the host", default="")
+    upload_id: int = Field(description="The unique id of the file")
```

Comment:
The formatting here is not great.

There is a pre-commit hook configured to run [Black](https://github.com/psf/black) on your code, but it doesn't seem to be working.

Install Black globally (`sudo pip install black`) and setup as File Watcher in Pycharm.

https://black.readthedocs.io/en/stable/integrations/editors.html#pycharm-intellij-idea